packages/apollo-codegen-core/src/__tests__/__snapshots__/jsonOutput.ts.snap

```diff
@@ -658,6 +692,265 @@ exports[`JSON output should generate JSON output for an input object type with d
+"{
+	\\"operations\\": [
+		{
+			\\"filePath\\": \\"GraphQL request\\",
+			\\"operationName\\": \\"HeroForEpisode\\",
+			\\"operationType\\": \\"query\\",
+			\\"rootType\\": \\"Query\\",
+			\\"variables\\": [],
+			\\"source\\": \\"query HeroForEpisode {\\\\n  hero(episode: JEDI) {\\\\n    __typename\\\\n    name\\\\n    ... on Droid {\\\\n      primaryFunction\\\\n    }\\\\n  }\\\\n}\\",
+			\\"fields\\": [
+				{
+					\\"responseName\\": \\"hero\\",
+					\\"fieldName\\": \\"hero\\",
+					\\"type\\": \\"Character\\",
+					\\"args\\": [
+						{
+							\\"name\\": \\"episode\\",
+							\\"value\\": \\"JEDI\\",
+							\\"type\\": \\"Episode\\"
+						}
+					],
+					\\"isConditional\\": false,
+					\\"isDeprecated\\": false,
+					\\"fields\\": [
+						{
+							\\"responseName\\": \\"__typename\\",
+							\\"fieldName\\": \\"__typename\\",
+							\\"type\\": \\"String!\\",
+							\\"isConditional\\": false,
+							\\"isDeprecated\\": false
+						},
+						{
+							\\"responseName\\": \\"name\\",
+							\\"fieldName\\": \\"name\\",
+							\\"type\\": \\"String!\\",
+							\\"isConditional\\": false,
+							\\"description\\": \\"The name of the character\\",
+							\\"isDeprecated\\": false
+						}
+					],
+					\\"fragmentSpreads\\": [],
+					\\"inlineFragments\\": [
+						{
+							\\"typeCondition\\": \\"Droid\\",
+							\\"possibleTypes\\": [
+								\\"Droid\\"
+							],
+							\\"fields\\": [
+								{
+									\\"responseName\\": \\"__typename\\",
+									\\"fieldName\\": \\"__typename\\",
+									\\"type\\": \\"String!\\",
+									\\"isConditional\\": false,
+									\\"isDeprecated\\": false
+								},
+								{
+									\\"responseName\\": \\"name\\",
+									\\"fieldName\\": \\"name\\",
+									\\"type\\": \\"String!\\",
+									\\"isConditional\\": false,
+									\\"description\\": \\"What others call this droid\\",
+									\\"isDeprecated\\": false
+								},
+								{
+									\\"responseName\\": \\"primaryFunction\\",
+									\\"fieldName\\": \\"primaryFunction\\",
+									\\"type\\": \\"String\\",
+									\\"isConditional\\": false,
+									\\"description\\": \\"This droid's primary function\\",
+									\\"isDeprecated\\": false
+								}
+							],
+							\\"fragmentSpreads\\": []
+						}
+					]
+				}
+			],
+			\\"fragmentSpreads\\": [],
+			\\"inlineFragments\\": [],
+			\\"fragmentsReferenced\\": [],
+			\\"sourceWithFragments\\": \\"query HeroForEpisode {\\\\n  hero(episode: JEDI) {\\\\n    __typename\\\\n    name\\\\n    ... on Droid {\\\\n      primaryFunction\\\\n    }\\\\n  }\\\\n}\\",
+			\\"operationId\\": \\"173a7ebf41abd014f3fc54ddd007f6182fca738bd44a1e2d1bb75503c8bf79f6\\"
+		}
+	],
+	\\"fragments\\": [],
+	\\"typesUsed\\": [],
+	\\"unionTypes\\": [],
+	\\"interfaceTypes\\": [
+		{
+			\\"name\\": \\"Character\\",
+			\\"types\\": [
+				\\"Human\\",
+				\\"Droid\\"
```

Comment:
Isn't there also an `Alien` type? Or is this going to only have the possible types that are actually used?

Reply:
@designatednerd The schema the tests are using does't have an `Alien` type. You can see it [here](https://github.com/apollographql/apollo-tooling/blob/f2c13e5e9d5a113369c5c236dc038986f4cb0872/__fixtures__/starwars/schema.json).